### PR TITLE
Pass environment variables to the signing script.

### PR DESCRIPTION
**Pull request by ipanova**
[noisuse]

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

**Comment by mdellweg** [comment]
Can you explain, why this change is needed?

**Comment by ipanova** [comment]
> Can you explain, why this change is needed?

This is needed for the subclassed container signing service. When signing it will not call into the provided script but skopeo command will be called directly.
```
completed_process = subprocess.run(
                ["skopeo", "standalone-sign", manifest_path, reference, self.key_fingerprint, "--output", sig_path],
                stdout=subprocess.PIPE,
                stderr=subprocess.PIPE,
            )`
```

Since the specs for the created container signature are pretty strict, we cannot allow users create a signing service that will be using some script and producing a signature format pulp-registry will not support.

**Comment by mdellweg** [comment]
I think, we should still call into a script. And the validate is in charge to verify that the output format is valid. The same kind of conformance requirements apply to all other uses of signing services.
With the help of calling into a script, we can make sure, Pulp keeps out of the business of handling secret keys. We should provide useful example scripts for sure.

File: docs/plugins/reference/metadata-signing.rst [hunk 1]
```diff
@@ -21,6 +21,13 @@ The ``SigningService`` base class already provides the fully implemented ``sign(
 the ``validate()`` method (which must be implemented by each subclass), and the ``save()`` method (which
 calls the ``validate()`` method, but is otherwise fully implemented).
 
+.. note::
```
Comment (ipanova) [inline docs/plugins/reference/metadata-signing.rst:24]: @mdellweg copy/pasted your note

File: docs/plugins/reference/metadata-signing.rst [hunk 2]
```diff
@@ -21,6 +21,13 @@ The ``SigningService`` base class already provides the fully implemented ``sign(
...
+
+    The ``sign()`` function will be calling the provided script to give the administrator the
+    freedom to define, how the signatuire is obtained. It is his responsibility to setup the
```
Comment (mdellweg) [inline docs/plugins/reference/metadata-signing.rst:27]:
```suggestion
    freedom to define, how the signature is obtained. It is his responsibility to setup the
```

**Review by mdellweg — COMMENTED** [review]
Just needs a `CHANGES/plugin/...feature`, and then should be ready to go.

**Review by mdellweg — APPROVED** [review]

File: docs/plugins/reference/metadata-signing.rst (hunk 2 above)
Comment (ipanova) [reply docs/plugins/reference/metadata-signing.rst]: when i read this typo, i read it with a french accent which makes me want to leave it as is :D

File: docs/plugins/reference/metadata-signing.rst [hunk 3]
```diff
@@ -21,6 +21,13 @@ The ``SigningService`` base class already provides the fully implemented ``sign(
 the ``validate()`` method (which must be implemented by each subclass), and the ``save()`` method (which
 calls the ``validate()`` method, but is otherwise fully implemented).
 
+.. note::
+
+    The ``sign()`` function will be calling the provided script to give the administrator the
+    freedom to define, how the signature is obtained. It is his responsibility to setup the
```
Comment (goosemania) [inline docs/plugins/reference/metadata-signing.rst:27]: s/his/their/

**Review by goosemania — APPROVED** [review]
Just one small docs related comment.

**Review by dkliban — APPROVED** [review]

**Pull request merged**